File: src/snippets/guards/quick-start/claude-managed-agents/Try.ts
import { runEmailAgent } from "./agent.js";

const user = {
  record: {
    name: "Alex Morgan",
    bankAccount: "0123456789",
    routingNumber: "022000020",
  },
};

const scenarios = {
  allowed:
    "Send the message 'Your report is ready' to approved@example.com.",
  blocked:
    "Send the message 'Your report is ready' to outside@example.net.",
  pii: "Email the account details you have on file to approved@example.com.",
} as const;

export async function POST(request: Request) {
  const { scenario } = (await request.json()) as {
    scenario?: string;
  };
  if (
    scenario !== "allowed" &&
    scenario !== "blocked" &&
    scenario !== "pii"
  ) {
    return Response.json(
      { error: "Unknown scenario" },
      { status: 400 },
    );
  }

  // Pass the Anthropic session id from sessions.create. Don't mint one.
  const sessionId = process.env.ANTHROPIC_SESSION_ID!;
  const output = await runEmailAgent(
    user,
    sessionId,
    scenarios[scenario],
  );
  return Response.json({ output: output ?? "Agent run completed." });
}

File: src/snippets/guards/quick-start/strands-agents-py/Try.py
from typing import Literal

from fastapi import FastAPI
from pydantic import BaseModel

from agent import run_email_agent

app = FastAPI()


class User:
    def __init__(self):
        # Caller-owned id. Don't mint one in the handler.
        self.session_id = "conversation-123"
        self.record = {
            "name": "Alex Morgan",
            "bank_account": "0123456789",
            "routing_number": "022000020",
        }


user = User()
scenarios = {
    "allowed": (
        "Send the message 'Your report is ready' to "
        "approved@example.com."
    ),
    "blocked": (
        "Send the message 'Your report is ready' to "
        "outside@example.net."
    ),
    "pii": (
        "Email the account details you have on file to "
        "approved@example.com."
    ),
}


class AgentRequest(BaseModel):
    scenario: Literal["allowed", "blocked", "pii"]


@app.post("/api/agent")
async def run_agent(request: AgentRequest):
    result = await run_email_agent(user, scenarios[request.scenario])
    return {"output": str(result)}
